### PR TITLE
Added a `defaults` package.

### PR DESCRIPTION
**Pull request by empijei**
For now it contains just a factory for safe ServeMuxConfigs.

### CODE REVIEW

File: safehttp/defaults/defaults.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/google/go-safeweb/safehttp/plugins/csp"
+	"github.com/google/go-safeweb/safehttp/plugins/fetchmetadata"
+	"github.com/google/go-safeweb/safehttp/plugins/hostcheck"
+	"github.com/google/go-safeweb/safehttp/plugins/hsts"
+	"github.com/google/go-safeweb/safehttp/plugins/staticheaders"
+	"github.com/google/go-safeweb/safehttp/plugins/xsrf/xsrfhtml"
+)
+
+// ServeMuxConfig creates a safe and ready to use ServeMuxConfig with all necessary security interceptors installed.
+func ServeMuxConfig(xsrfKey string, hosts, allowedCORSOrigins []string) *safehttp.ServeMuxConfig {
```
Comment (kele) [inline safehttp/defaults/defaults.go:31]: Maybe this could benefit from go/gotip/episodes/41?

**Review by kele — APPROVED** [review]

File: safehttp/defaults/defaults.go (hunk 1 above)
Comment (empijei) [reply safehttp/defaults/defaults.go:31]: I thought about this but solution 2 doesn't really fit, and if you adopt solution 1 and you add one more parameter every user of this function will keep working without realizing the API has changed. With that solution there is no way to make parameters mandatory.

Comment (kele) [reply safehttp/defaults/defaults.go:31]: Yup, you're right.

In this case, it seems to be more natural to me to have them in the `hosts, xsrf, cors` order. Alternatively, just `hosts, xsrf` and simply not support CORS in the default mux. WDYT?

File: safehttp/defaults/defaults_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			form, err := r.URL.Query()
+			if err != nil {
+				t.Errorf("Cannot parse GET form: %v", err)
+			}
+			b := form.Bool("test", false)
+			if !b {
+				t.Error("test parameter, got false, want true")
+			}
+			w.Write(safehtml.HTMLEscaped(resp))
+			return safehttp.Result{}
```
Comment (kele) [inline safehttp/defaults/defaults_test.go:41]:
```suggestion
			return w.Write(safehtml.HTMLEscaped(resp))
```

**Comment by mikue** [comment]
Since kele requested review;
A safe-by-default `ServeMuxConfig` should definitely live somewhere. I'm not super convinced of using a "defaults" package vs. e.g. having a `safehttp.NewServeMuxConfig`. Do you have ideas on what else would live in the defaults package eventually?

I also wonder if we should do more checks for the input to this function since it will be the main entry point for end-users. I suppose usage of this function will be in scope for security review anyway though, so maybe we don't need to be so strict (e.g. checking for an empty xsrf key).

**Review by mikue — APPROVED** [review]

**Comment by kele** [comment]
> A safe-by-default `ServeMuxConfig` should definitely live somewhere. I'm not super convinced of using a "defaults" package vs. e.g. having a `safehttp.NewServeMuxConfig`. Do you have ideas on what else would live in the defaults package eventually?

Not possible. Dependency cycles.

> I also wonder if we should do more checks for the input to this function since it will be the main entry point for end-users. I suppose usage of this function will be in scope for security review anyway though, so maybe we don't need to be so strict (e.g. checking for an empty xsrf key).

+1 on this

File: safehttp/defaults/defaults.go (hunk 1 above)
Comment (empijei) [reply safehttp/defaults/defaults.go:31]: Agreed!

**Comment by kele** [comment]
Ready to merge. Go ahead @empijei

**Pull request merged**